### PR TITLE
Add support to CLI for pre-building flows

### PR DESCRIPTION
**Pull request by jcrist**
This PR:

- Adds a new `prefect build` CLI command for building flows and creating/updating a `.json` file containing their representations.
- Adds a `--json` flag to `prefect register` for registering flows from a `.json` file. Note that this file may be local, or hosted elsewhere (currently only `https`/`http`, `s3`, and `gcs` are supported).

Combined, these let users build flows once and host the `json` results somewhere for later registration into one or more projects.

For example, you might build your flows as part of CI (and host the output `flows.json` file on github pages as part of your docs). Users can then register flows from your repo as:

```
$ prefect register --json https://username.github.io/your-project/flows.json --project their-project-name
```

With pre-built flows, the client doesn't need access to the source or dependencies, making registration very lightweight.

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/cli/build_register.py [hunk 1]
```diff
@@ -0,0 +1,972 @@
...
+from prefect.utilities.filesystems import read_bytes_from_path, parse_path
+from prefect.utilities.graphql import with_args, EnumValue, compress
+from prefect.storage import Local, Module
+from prefect.run_configs import UniversalRun
+
+
+FlowLike = Union[box.Box, "prefect.Flow"]
+
+
+class SimpleFlowSchema(marshmallow.Schema):
```
Comment (cicdw) [inline src/prefect/cli/build_register.py:29]: Any reason we shouldn't use pydantic here, or is it not worth the new dependency?

File: src/prefect/cli/build_register.py [hunk 2]
```diff
@@ -0,0 +1,972 @@
...
+        "passed multiple times to specify multiple modules to register."
+    ),
+    multiple=True,
+)
+@click.option(
+    "--json",
+    "-j",
+    "json_paths",
+    help=(
+        "A path or URL to a json file containing the flow(s) to register. "
```
Comment (cicdw) [inline src/prefect/cli/build_register.py:657]: The fact that the schema of this file is not a dump of `flow.serialize()` took me a minute to work through and seems like something we should call out here -- not sure of the simplest way to say "we expect this file to be the output of `prefect build`

**Review by cicdw — CHANGES_REQUESTED** [review]
Two questions, overall this looks great

File: src/prefect/cli/build_register.py (hunk 1 above)
Comment (jcrist) [reply src/prefect/cli/build_register.py:29]: I want to use pydantic, but sticking with marshmallow for now. We should maybe prioritize transitioning though.

File: src/prefect/cli/build_register.py (hunk 2 above)
Comment (jcrist) [reply src/prefect/cli/build_register.py:657]: Sure, I can expand the help string here.

Comment (jcrist) [reply src/prefect/cli/build_register.py]: Done

**Review by cicdw — APPROVED** [review]

**Pull request merged**